src/process/LokinetProcessManager.cpp

```diff
@@ -12,7 +16,10 @@ constexpr auto MANAGED_KILL_WAIT = 5s;
 LokinetProcessManager::LokinetProcessManager()
     : m_managedThreadRunning(false)
     , m_didLaunchProcess(false)
-{
+{
+#ifdef Q_OS_WIN
+    ::CreateMutex(nullptr, FALSE, "lokinet_qt5_ui");

```

Comment:
I'm confused, is there some magical side effect of `::CreateMutex()` that isn't obvious? I wouldn't think that creating a mutex would somehow have anything to do with an installer or displaying a dialog to the user, etc.

Also, this should probably go in the Windows-specific ProcessManager class.

Reply:
the installer package tries to gain ownership of the named mutex object separately. if the ui is running in the background, this prevents the installer from continuing until the ui and/or daemon are closed

the mutex is released upon process termination, and the installer ultimately destroys them.

I wanted to place this somewhere as close to the start of the UI process as possible